tests/test_source_sampling.py

```diff
@@ -1673,21 +1714,27 @@ def _source_sampling_test_template(mode, cell_fracs_list, src_tag,
 
     # construct Sampler
     sampler = Sampler(filename, tag_names, e_bounds, mode)
+    # remove the temporary file
+    os.remove(filename)
 
     # sampling and tally, tally should be defined by the mesh cell_fracs
     num_samples = 5000
     particles = []
 
     seed(1953)
     for i in range(num_samples):
-        s = sampler.particle_birth(np.array([uniform(0, 1) for x in range(6)]))
+        rands = np.array([uniform(0, 1) for x in range(6)])
+        s = sampler.particle_birth(rands)
         # check w, and c for each particle
         # calculate the expected weight and cell_number
         exp_w, exp_c = _cal_exp_w_c(s, mode, cell_fracs, src_tag, bias_tag)
         assert_equal(s.w, exp_w)
         # when mode in (0, 1, 2), the set exp_c is (-1), otherwise it contains
         # several available cell number
-        assert(s.c in exp_c)
+        if mode in (0, 1, 2):
```

Comment:
I feel like we run into this mode check a lot, where we're really just trying to figure out if we're in sub-voxel or voxel mode. Didn't we discuss having two "mode" enumerators at some point? One for the sampling type (analog, uniform, user-specified) and another for the sampling regions (voxel, sub-voxel). My terminology may not be ideal here..

Reply:
Yes, we have discussed that `mode` before. And I added that at a PR, however, I removed that variable at a recent PR because I was trying to unify the voxel/sub-voxel source sampling.
It turns out that I need to add that enumerator back.